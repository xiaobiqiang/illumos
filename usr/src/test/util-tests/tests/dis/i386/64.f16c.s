/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test F16C related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vcvtph2ps	%xmm0, %xmm1
	vcvtph2ps	(%rax), %xmm1
	vcvtph2ps	0x24(%rax), %xmm1
	vcvtph2ps	%xmm0, %ymm1
	vcvtph2ps	(%rax), %ymm1
	vcvtph2ps	0x24(%rax), %ymm1

	vcvtps2ph	$0x10, %xmm0, %xmm1
	vcvtps2ph	$0x10, %xmm3, (%rbx)
	vcvtps2ph	$0x10, %xmm4, 0x10(%rcx)
	vcvtps2ph	$0x10, %ymm0, %xmm1
	vcvtps2ph	$0x10, %ymm3, (%rbx)
	vcvtps2ph	$0x10, %ymm4, 0x10(%rcx)
.size libdis_test, [.-libdis_test]
