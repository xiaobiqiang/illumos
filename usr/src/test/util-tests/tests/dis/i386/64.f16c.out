    libdis_test:      c4 e2 79 13 c8     vcvtph2ps %xmm0,%xmm1
    libdis_test+0x5:  c4 e2 79 13 08     vcvtph2ps (%rax),%xmm1
    libdis_test+0xa:  c4 e2 79 13 48 24  vcvtph2ps 0x24(%rax),%xmm1
    libdis_test+0x10: c4 e2 7d 13 c8     vcvtph2ps %xmm0,%ymm1
    libdis_test+0x15: c4 e2 7d 13 08     vcvtph2ps (%rax),%ymm1
    libdis_test+0x1a: c4 e2 7d 13 48 24  vcvtph2ps 0x24(%rax),%ymm1
    libdis_test+0x20: c4 e3 79 1d c1 10  vcvtps2ph $0x10,%xmm0,%xmm1
    libdis_test+0x26: c4 e3 79 1d 1b 10  vcvtps2ph $0x10,%xmm3,(%rbx)
    libdis_test+0x2c: c4 e3 79 1d 61 10  vcvtps2ph $0x10,%xmm4,0x10(%rcx)
                      10 
    libdis_test+0x33: c4 e3 7d 1d c1 10  vcvtps2ph $0x10,%ymm0,%xmm1
    libdis_test+0x39: c4 e3 7d 1d 1b 10  vcvtps2ph $0x10,%ymm3,(%rbx)
    libdis_test+0x3f: c4 e3 7d 1d 61 10  vcvtps2ph $0x10,%ymm4,0x10(%rcx)
                      10 
